packages/shared/src/hooks/useReadingRank.ts

```diff
@@ -117,30 +117,58 @@ export default function useReadingRank(
       neverShowRankModal,
     );
 
-  const updateShownProgress = async () => {
+  const unmountRef = useRef(false);
+  const timeoutRef = useRef<number>();
+  const visibilityRef = useRef(null);
+
+  const updateShownProgress = useCallback(() => {
+    if (unmountRef) {
```

Comment:
I think you originally meant this
```suggestion
    if (unmountRef?.current) {
```